cmd/jujud/agent/machine/manifolds.go

```diff
@@ -784,6 +786,14 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 				Logger:        loggo.GetLogger("juju.worker.secretbackendsrotate"),
 			},
 		))),
+
+		// The controlsocket worker runs on the controller machine.
+		controlSocketName: ifController(controlsocket.Manifold(controlsocket.ManifoldConfig{
```

Comment:
I believe `ifController` is already implied by a dependency on `stateName`.

Reply:
you're right, but does it hurt to be explicit here?

Reply:
The use of the `ifController` flag here prevents the dependency engine from even considering the worker eligible to be started right? Without it, the engine is burdened with waiting for a `stateName` dependency that is never realised?

Reply:
@wallyworld I can't quite parse your comment. Are you for or against the `ifController`? It doesn't seem to make a functional difference either way.

Reply:
I am happy with `ifController` being here, it shows explicit intention.